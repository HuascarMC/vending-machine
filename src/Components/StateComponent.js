import React from 'react';

class StateComponent extends React.Component {
  render() {
    return(
      <div className="state-wrapper">
        <p className="title">state</p>
        <p className="state-text">In Stock: {JSON.stringify(this.props.state.stockInventoryState) } </p>
        <p className="state-text">Enough change: {JSON.stringify(this.props.state.coinInventoryState) } </p>
        <p className="state-text">Enough balance: {JSON.stringify(this.props.state.balanceState) } </p>
      </div>
    );
  }
}

export default StateComponent;
