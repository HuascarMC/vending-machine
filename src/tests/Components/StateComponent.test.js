import React from 'react';
import StateComponent from '../../Components/StateComponent'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

var state = {
  coinInventoryState: true,
  stockInventoryState: true,
  balanceState: true
}

it('should render without crashing', () => {
  shallow(<StateComponent state={ state } />);
});

it('should render state-wrapper div', () => {
  const wrapper = shallow(<StateComponent state={ state} />);
  const div = wrapper.find('.state-wrapper');
  expect(div.exists()).toEqual(true);
});

it('should render p tag with text state', () => {
  const wrapper = shallow(<StateComponent state={ state} />);
  const p = wrapper.find('p').at(0);
  expect(p.text()).toEqual('state');
});

it('should render p tag with title selector', () => {
  const wrapper = shallow(<StateComponent state={ state} />);
  const p = wrapper.find('.title');
  expect(p.exists()).toEqual(true);
});
