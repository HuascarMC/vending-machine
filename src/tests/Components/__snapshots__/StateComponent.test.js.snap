// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="state-wrapper"
>
  <p
    className="title"
  >
    state
  </p>
  <p
    className="state-text"
  >
    In Stock: 
    true
     
  </p>
  <p
    className="state-text"
  >
    Enough change: 
    true
     
  </p>
  <p
    className="state-text"
  >
    Enough balance: 
    true
     
  </p>
</div>
`;
